components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/service/WebSubTopicManagerImpl.java

```diff
@@ -128,24 +131,82 @@ private void makeTopicMgtAPICall(String topic, String webSubHubBaseUrl, String o
             final long requestStartTime = System.currentTimeMillis();
 
             try (CloseableHttpResponse response = (CloseableHttpResponse) clientManager.execute(httpPost)) {
-                int responseCode = response.getStatusLine().getStatusCode();
-                if (responseCode >= 500 && attempt < MAX_RETRIES) {
+
+                StatusLine statusLine = response.getStatusLine();
+                int responseCode = statusLine.getStatusCode();
+                String responsePhrase = statusLine.getReasonPhrase();
+
+                if (responseCode >= 500 && responseCode < 600 && attempt < MAX_RETRIES) {
                     attempt++;
-                    log.info("Retrying topic management API call, attempt " + attempt + " for topic: " + topic);
-                    try {
-                        Thread.sleep(RETRY_DELAY_MS);
-                    } catch (InterruptedException ie) {
-                        Thread.currentThread().interrupt();
-                        throw handleServerException(ERROR_REGISTERING_HUB_TOPIC, ie, topic, tenantDomain);
-                    }
+                    log.debug("Retrying topic management API call, attempt " + attempt + " for topic: " + topic);
+                    Thread.sleep(RETRY_DELAY_MS);
                     continue;
                 }
-                handleTopicMgtResponse(response, httpPost, topic, operation, requestStartTime);
-                break; // Success or handled error
-            } catch (IOException | WebSubAdapterException e) {
+
+                if (responseCode == HttpStatus.SC_OK) {
+                    HttpEntity entity = response.getEntity();
+                    WebSubHubCorrelationLogUtils.triggerCorrelationLogForResponse(httpPost, requestStartTime,
+                            WebSubHubCorrelationLogUtils.RequestStatus.COMPLETED.getStatus(),
+                            String.valueOf(responseCode), responsePhrase);
+                    if (entity != null) {
+                        String responseString = EntityUtils.toString(entity, StandardCharsets.UTF_8);
+                        if (RESPONSE_FOR_SUCCESSFUL_OPERATION.equals(responseString)) {
+                            log.debug("Success WebSub Hub operation: " + operation + ", topic: " + topic);
+                        } else {
+                            throw handleServerException(ERROR_INVALID_RESPONSE_FROM_WEBSUB_HUB, null, topic,
+                                    operation, responseString);
+                        }
+                    } else {
+                        String message =
+                                String.format(ERROR_EMPTY_RESPONSE_FROM_WEBSUB_HUB.getDescription(), topic, operation);
+                        throw new WebSubAdapterServerException(message, ERROR_EMPTY_RESPONSE_FROM_WEBSUB_HUB.getCode());
+                    }
```

Comment:
Try to remove these if else branches

Reply:
Will be address via https://github.com/wso2-extensions/identity-event-publishers/pull/25/commits/67d32091145f93afc04a6815cd9e2ef1828a00c3